src/plugins/data/public/ui/filter_bar/filter_editor/value_input_type.test.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import { ValueInputType } from './value_input_type';
+
+let onChangeMock: any;
+
+describe('Value input type', () => {
+  beforeAll(() => {
+    onChangeMock = jest.fn();
+  });
+  it('is number', async () => {
+    const valueInputProps = {
+      value: 1,
+      type: 'number',
+      onChange: onChangeMock,
+      onBlur: () => {},
+      placeholder: '',
+    };
+    const component = mountWithIntl(<ValueInputType {...valueInputProps} />);
+    expect(component).toMatchSnapshot();
+  });
+
+  it('is number bigint', async () => {
+    const valueInputProps = {
+      value: '1n',
+      type: 'number',
+      onChange: onChangeMock,
+      onBlur: () => {},
+      placeholder: '',
+    };
+    const component = mountWithIntl(<ValueInputType {...valueInputProps} />);
+    expect(component).toMatchSnapshot();
```

Comment:
Can we not use snapshot tests here? Taking a look at the snapshot and they dont offer much value here

Reply:
I think they do provide values, the unit tests are mainly testing if `switch (this.props.type)` are rendering correct components based on the input type. For example, string type should render <EuiFieldText> with correct placeholder, boolean type should render <EuiSelect> with correct selections.

```
switch (this.props.type) {
      case 'string':
        inputElement = (
          <EuiFieldText
           ...
          />
        );
        break;
      case 'number':
        inputElement = (
          <EuiFieldNumber
            ...
          />
        );
        break;
      case 'date':
        inputElement = (
          <EuiFieldText
            ...
          />
        );
        break;
      case 'ip':
        inputElement = (
          <EuiFieldText
           ...
          />
        );
        break;
      case 'boolean':
        inputElement = (
          <EuiSelect
            fullWidth={this.props.fullWidth}
            options={[...]}
            ...
          />
        );
```

Reply:
i meant can we just snapshot the value we expect instead of the whole thing. If this test breaks, it should be only because of the changes related to what you are testing. Right now, if the component chnages, these tests will break too. Keep unit tests scoped to just test for just the feature and not anything else. Otherwise when things change, a lot of these test will break.